packages/core/src/utils-hoist/worldwide.ts

```diff
@@ -99,9 +65,20 @@ export const GLOBAL_OBJ = globalThis as unknown as InternalGlobal;
  * @param obj (Optional) The global object on which to look for `__SENTRY__`, if not `GLOBAL_OBJ`'s return value
  * @returns the singleton
  */
-export function getGlobalSingleton<T>(name: keyof SentryCarrier, creator: () => T, obj?: unknown): T {
-  const gbl = (obj || GLOBAL_OBJ) as InternalGlobal;
-  const __SENTRY__ = (gbl.__SENTRY__ = gbl.__SENTRY__ || {});
+export function getGlobalSingleton<Prop extends keyof SentryCarrier>(
```

Comment:
Made this more type safe and actually inferring types correctly for passed in stuff!